src/api/app.py

```diff
@@ -260,16 +260,29 @@ async def log_out(request: Request, session: SessionDeletePydantic):
 #     return Response(status_code=501)
 #     #return event_controller.add_event(json.loads(request.data))
 #
-# @app.post('/api/user/course')
-# async def add_student_course(request: Request, credentials: UserCoursePydantic):
-#     if 'user' not in request.session:
-#         return Response("Not authorized", status_code=403)
-#     resp, error = course_select.add_selection(credentials.name, credentials.semester, credentials.cid)
-#     return Response(status_code=200) if not error else Response(error, status_code=500)
-
-@app.delete('/api/user/course')
-async def remove_student_course(request: Request, courseDelete:CourseDeletePydantic):
+@app.post('/api/user/course')
+async def add_student_course(request: Request, credentials: UserCoursePydantic):
     if 'user' not in request.session:
         return Response("Not authorized", status_code=403)
-    resp,error = course_select.remove_selection(courseDelete.name, courseDelete.semester, request.session['user']['user_id'], courseDelete.cid)
+    resp, error = await course_select.add_selection(credentials.name, credentials.semester, credentials.cid)
     return Response(status_code=200) if not error else Response(error, status_code=500)
+#
+#
```

Comment:
You need to finish/uncomment the delete and get methods. It also appears that there was some conflicts here as you have outdated versions of the methods as they are still using flask syntax in the commented versions. So make sure you get the correct version from TortoiseConversion when you do uncomment.